app/build.gradle.kts

```diff
@@ -143,6 +147,20 @@ android {
     buildFeatures {
         buildConfig = true
     }
+    flavorDimensions += "store"
+    productFlavors {
+        create("gplay") {
+            dimension = "store"
+            isDefault = true
+            buildConfigField("String", "SHORT_FLAVOR_DESCRIPTION", "\"G\"")
+            buildConfigField("String", "FLAVOR_DESCRIPTION", "\"GooglePlay\"")
```

Comment:
Maybe you can remove the firebase distribution plugin from above and do something like this here:

```kotlin
create("gplay") {
            ...
            apply(plugin = libs.plugins.firebaseAppDistribution.get().pluginId)
            buildTypes.getByName("nightly") {
                project.extensions.configure<AppDistributionExtension>("firebaseAppDistribution") {
                        artifactType = "APK"
                        // We upload the universal APK to fix this error:
                        // "App Distribution found more than 1 output file for this variant.
                        // Please contact firebase-support@google.com for help using APK splits with App Distribution."
                        artifactPath = "$rootDir/app/build/outputs/apk/nightly/app-universal-nightly.apk"
                        // artifactType = "AAB"
                        // artifactPath = "$rootDir/app/build/outputs/bundle/nightly/app-nightly.aab"
                        // This file will be generated by the GitHub action
                        releaseNotesFile = "CHANGES_NIGHTLY.md"
                        groups = "external-testers"
                        // This should not be required, but if I do not add the appId, I get this error:
                        // "App Distribution halted because it had a problem uploading the APK: [404] Requested entity was not found."
                        appId = "1:912726360885:android:e17435e0beb0303000427c"
                    }
            }
        }
```

Reply:
Actually, there's no need for the whole `configure...` workaround, it seems like just using `firebaseAppDistribution { ... }` instead should work.

Reply:
Actually, my bad, I think this actually still applies it to all the variants... I would probably work if we added the flavors using `register` as that's lazy loaded, but that doesn't seem to be possible.

Reply:
In fact, the plugin seems to automatically apply itself to all flavors, so there's probably no way to improve this in any way:

```java
androidExt.getProductFlavors().all((flavor) -> {
    BuildVariantHelper.getExtensions(flavor).add("firebaseAppDistribution", AppDistributionExtension.class);
});
```